### ISSUE TITLE
bdb.open: Failed to open a database at data/paperdoll_exemplars.bdb: Invalid argument

### ISSUE THREAD

**Issue by xlpiao** [kyamagu/matlab-bdb#1]
I need some help for running paperdoll.

When I run:
load data/paperdoll_pipeline.mat config;
input_image = imread('./fashionista/001.jpg');
input_sample = struct('image', imencode(input_image, 'jpg'));

result = feature_calculator.apply(config, input_sample);    %error

The result comes: 
[08-Jun-2015 21:12:03] Error:feature_calculator: 1 / 1
Error using mex_function_
Failed to open a database at data/paperdoll_exemplars.bdb: Invalid argument

Error in bdb.open (line 72)
  id = mex_function_(mfilename, filename, varargin{:});

Error in softmask_transferer.apply>load_exemplars (line 83)
    database_id = bdb.open(config.database_file);

Error in softmask_transferer.apply>process_sample (line 45)
  [exemplars, labels] = load_exemplars(config, sample);

Error in softmask_transferer.apply (line 34)
    [softmask_transfer, labels] = process_sample(config, sample);

Error in feature_calculator.apply (line 25)
        sample = calculators{j}(config{j}, sample, varargin{:}, 'Encode', false);

It seems that these errors come from matlab-bdb.  
I finally found that the error message comes from database_id = bdb.open(config.database_file, 'Rdonly', true, 'Create', false);

Is there any solution to fix it ?

While platforms are: 
64 bit
ubuntu 12.04
Matlab R2012a

**Comment by kyamagu** [comment]
@xlpiao do you have the bdb file? Make sure that there is a file at `config.database_file`

**Comment by xlpiao** [comment]
@kyamagu 
yes, I have.

>> config.database_file

ans =

data/paperdoll_exemplars.bdb

paperdoll_exemplars.bdb is downloaded from http://vision.is.tohoku.ac.jp/~kyamagu/research/paperdoll/model.*

**Comment by kyamagu** [comment]
@xlpiao and your current directory is at `paperdoll`, right?

**Comment by xlpiao** [comment]
@kyamagu

Yes!

**Comment by kyamagu** [comment]
@xlpiao hmm, could be a platform issue. Can you use the bdb API in other locations? Like,

```
filename = 'test.bdb';
bdb.open(filename, 'Rdonly', false, 'Create', true);
bdb.put('foo', 'bar');
bdb.get('foo');
bdb.close();
```

**Comment by xlpiao** [comment]
@kyamagu

It works well ~

>> filename = 'test.bdb';
bdb.open(filename, 'Rdonly', false, 'Create', true);
bdb.put('foo', 'bar');
bdb.get('foo');
bdb.close();
>> filename

filename =

test.bdb

**Comment by xlpiao** [comment]
@kyamagu

However, it still not work in that paperdoll

**Comment by kyamagu** [comment]
@xlpiao Perhaps the database binary is damaged. Could be a failure in download, but I'm not sure...

**Comment by xlpiao** [comment]
@xlpiao 
Yes, I actually have been trying to regenerate the paperdoll_exemplars.bdb

**Comment by kyamagu** [comment]
@xlpiao Try md5sum

```
for i in 00 01 02 03 04 05 06 07 08 09 10 11 12 13 14; do md5sum -b models-v1.0.tar.$i; done
3f14f5d90e4c3c3ce014311dce0df1bf *models-v1.0.tar.00
46bb5d046dc6f9a6e6cb3c9832ab4c6d *models-v1.0.tar.01
85f089dd4a589e02fe5da1fb16b7dbae *models-v1.0.tar.02
b0f0d18bd9ec13fbc6c63e0a1fd6356d *models-v1.0.tar.03
1b7838c2d4c8287f900992f3e7969f9c *models-v1.0.tar.04
5e7f9c7a87e3cc753b4508daa65c247a *models-v1.0.tar.05
e7ae269f42e1b7bdf30f9cac3b7ea62a *models-v1.0.tar.06
96c92e94ae179fd805f731da65636604 *models-v1.0.tar.07
b3c5f7a89a78a7dc60ee57641b6297e9 *models-v1.0.tar.08
0371ddec6c5ce04cf185f30cfd8e92ce *models-v1.0.tar.09
e9b7a90856b58d7d47f5f28902ccc561 *models-v1.0.tar.10
6ced6bf6292c3893cc4ba429ac4617b8 *models-v1.0.tar.11
57d4b0617d984c767b4617da2e44158f *models-v1.0.tar.12
1ee83b90fd49b0fe4310c89ceaf69a17 *models-v1.0.tar.13
7db0e3291730e53ffed526144c2c8e10 *models-v1.0.tar.14
```

**Comment by hosnasattar** [comment]
@xlpiao  I do have the same problem. were you able to solve the problem?

**Comment by kyamagu** [comment]
@hosna This could be yet another version issue.

http://effbot.org/pyfaq/i-tried-to-open-berkeley-db-file-but-bsddb-produces-bsddb-error-22-invalid-argument-help-how-can-i-restore-my-data.htm

You can use `db-util` package of Ubuntu to check the database content in command line.
